beacon-chain/sync/batch_verifier.go

```diff
@@ -161,7 +161,7 @@ func (s *Service) validateWithKzgBatchVerifier(ctx context.Context, dataColumns
 
 	timeout := time.Duration(params.BeaconConfig().SecondsPerSlot) * time.Second
 
-	resChan := make(chan error)
+	resChan := make(chan error, 1)
```

Comment:
I don't actually think this needs to be buffered, because that could obfuscate issues. The other side of the channel should be ready to immediately drain it.

Reply:
Chatted with Terence offline and he reminded me that this channel is unique to the caller, so it would be fine to buffer like this, let the channel go out of scope and be garbage collected. I will confirm that there are no other references to the channel that could cause it to leak, otherwise I take it back, this is a nice simple solution.